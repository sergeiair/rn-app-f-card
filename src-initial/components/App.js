import React, { Component } from 'react';
import { useStrict } from 'mobx';
import { observer } from 'mobx-react/native';
import { enableLogging } from 'mobx-logger';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';

import Timer from './Timer';
import TimerStore from '../stores/TimerStore';

useStrict(true);

enableLogging({
  predicate: () => __DEV__ && Boolean(window.navigator.userAgent),
  action: true,
  reaction: false,
  compute: false
});

/**
 * @author winterbe
 */
@observer
export default class App extends Component {
  static defaultProps = {
    store: new TimerStore()
  };

  render() {
    const {store} = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.header}>
          <Text style={styles.headline}>
            RNTimerExample
          </Text>
        </View>
        <View style={styles.content}>
          {store.timers.map((timer, i) => (
            <Timer timerModel={timer} key={i}/>
          ))}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#FFF'
  },
  header: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  headline: {
    fontSize: 30
  },
  content: {
    flex: 5
  }
});
