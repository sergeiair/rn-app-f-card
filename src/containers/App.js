import React, { Component } from 'react';
import { Router, Scene } from 'react-native-router-flux';
import { StyleSheet } from 'react-native';
import { useStrict } from 'mobx';
import I18n from 'react-native-i18n';

import LocalizationService from './../services/localization.service';

import HomeContainer from './home/home.container';
import QuestionsContainer from './questions/questions.container';
import CardsContainer from './cards/cards.container';
import FishesContainer from './fishes/fishes.container';
import RulesContainer from './rules/rules.container';
import RuleContentContainer from './ruleContent/ruleContent.container';
import PlacesContainer from './places/places.container';
import FishContainer from './fish/fish.container';
import SettingsContainer from './settings/settings.container';
import TacklesContainer from './tackles/tackles.container';


import colors from '../core-styles/colors';

import '../core-styles/rk-styles';

useStrict(true);


export default class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      localeLoaded: false
    }
  }

  componentDidMount() {
	  LocalizationService
      .init()
        .finally(() => {
          this.setState({localeLoaded: true});
        })
  }

  render() {
    const {localeLoaded} = this.state;

    if (!localeLoaded) {
      return null;
    }

    return (
      <Router navigationBarStyle={styles.navBar}
        titleStyle={styles.navBarTitle}
        barButtonIconStyle={styles.barButtonIconStyle}
        duration={0}>
          <Scene key="root">
            <Scene key="home" component={HomeContainer} initial={true} title={I18n.t('texts.titles.welcome')}  />
            <Scene key="cards" component={CardsContainer} title={I18n.t('texts.titles.cards')}/>
            <Scene key="questions" component={QuestionsContainer} title={I18n.t('texts.titles.questions')} />
            <Scene key="fishes" component={FishesContainer} title={I18n.t('texts.titles.fishes')} />
            <Scene key="rules" component={RulesContainer} title={I18n.t('texts.titles.rules')} />
            <Scene key="places" component={PlacesContainer} title={I18n.t('texts.titles.places')}/>
            <Scene key="fish" component={FishContainer}/>
            <Scene key="ruleContent" component={RuleContentContainer}/>
            <Scene key="settings" component={SettingsContainer} title={I18n.t('texts.titles.settings')}/>
	          <Scene key="tackles" component={TacklesContainer} title={I18n.t('texts.titles.tackles')}/>
          </Scene>
      </Router>
    );
  }
}

const styles = StyleSheet.create({
	navBar: {
		backgroundColor: colors.blue,
		borderBottomColor: colors.blue,
		height: 55,
		borderBottomWidth: 1
	},
	navBarTitle: {
		fontSize: 17,
		lineHeight: 26,
    textAlign: 'center',
		color: colors.lightTone
	},
	barButtonIconStyle: {
		marginLeft: 15,
		tintColor: colors.lightTone
	}
});

