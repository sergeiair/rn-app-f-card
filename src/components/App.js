import React, { Component } from 'react';
import { Router, Scene } from 'react-native-router-flux';
import { useStrict } from 'mobx';
import { enableLogging } from 'mobx-logger';

import { HomeComponent } from './components';

useStrict(true);

enableLogging({
  predicate: () => __DEV__ && Boolean(window.navigator.userAgent),
  action: true,
  reaction: false,
  compute: false
});

export default class App extends Component {
  render() {
    return (
      <Router>
        <Scene key="root">
          <Scene key="home" component={ HomeComponent } title="Home" initial={true} />
        </Scene>
      </Router>
    );
  }
}

