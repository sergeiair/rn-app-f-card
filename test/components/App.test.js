import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../src/components/App';

it('renders <App/> correctly', () => {
    const component = renderer.create(
        <App/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    component.unmount();
});
