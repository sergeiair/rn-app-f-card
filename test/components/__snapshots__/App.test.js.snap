exports[`test renders <App/> correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 30,
        }
      }>
      RNTimerExample
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#222",
            "fontFamily": "Courier New",
            "fontSize": 26,
            "width": 160,
          }
        }>
        00:00,000
      </Text>
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Array [
              Object {},
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#0C42FD",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#337ab7",
                },
              ]
            }>
            Start
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
